runtime/README.md

```diff
@@ -692,107 +787,631 @@ Command Code: `0x434B_4558` ("CKEX")
+| iv size         | u32      | 12 (first message) or 0 (subsequent)     |
+| iv              | u8[...]  |                                          |
+| aad size        | u32      | Additional authenticated data size       |
+|                 |          | Must be 0 if this is not the first block |
+| aad             | u8[...]  | Additional authenticated data            |
+| ciphertext size | u32      |                                          |
+| ciphertext      | u8[...]  | Data to decrypt                          |
+
+*Table: `CM_AES_GCM_DECRYPT` output arguments*
+| **Name**       | **Type** | **Description**                           |
+| -------------- | -------- | ----------------------------------------- |
+| chksum         | u32      |                                           |
+| fips_status    | u32      | FIPS approved or an error                 |
+| context size   | u32      | Can be 0 if this is the final message     |
+| context        | u8[...]  |                                           |
+| plaintext size | u32      | Can be 0 if this is not the final message |
+| plaintext      | u8[...]  |                                           |
+
+The encrypted and authenticated context's internal structure will be:
+
+*Table: `CM_AES_GCM_DECRYPT` internal context*
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| tag size        | u32      |                 |
+| last length     | u32      |                 |
+| last iv         | u8[16]   |                 |
+| last GHASH      | u8[16]   |                 |
+| last ciphertext | u8[16]   |                 |
+
+
+### CM_ECDH_GENERATE
+
+This computes the first half of an Elliptic Curve Diffie-Hellman exchange to compute an ephemeral shared key pair with another party.
+
+Currently only supports the NIST P-384 curve.
+
+The returned context must be passed to the `CM_ECDH_FINISH` command. The context contains the (encrypted) secret coefficient.
+
+Command Code: `0x434D_4547` ("CMEG")
+
+*Table: `CM_ECDH_GENERATE` input arguments*
+| **Name**    | **Type** | **Description**      |
+| ----------- | -------- | -------------------- |
+| chksum      | u32      |                      |
+| curve/flags | u32      | Must be 0. Reserved. |
+
+*Table: `CM_ECDH_GENERATE` output arguments*
+| **Name**           | **Type** | **Description**                       |
+| ------------------ | -------- | ------------------------------------- |
+| chksum             | u32      |                                       |
+| fips_status        | u32      | FIPS approved or an error             |
+| context size       | u32      | size of context                       |
+| context            | u8[...]  | Used as the input to `CM_ECDH_FINISH` |
+| exchange data size | u32      | size of data to send in the exchange  |
+| exchange data      | u8[...]  | i.e., the public point                |
```

Comment:
Seems like we should document the format of this point? Is it 0x04 || x || y?

Reply:
I'm not even sure that `0x04` is necessary. I don't see it in the TLS exchange data at least. I'll document that.